### PR TITLE
Update to use --json-version=2

### PR DESCRIPTION
**Pull request by nwoltman**
The new JSON format is much more consistent and has fewer oddities than the old format.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -10,10 +10,12 @@
     "es6": true
   },
   "plugins": [
+    "flowtype",
     "prettier"
   ],
   "rules": {
     "consistent-return": "off",
+    "flowtype/define-flow-type",
```
Comment (amilajack) [inline .eslintrc]: The CI test is failing because of this one: https://travis-ci.org/amilajack/eslint-plugin-flowtype-errors/jobs/346625539#L502

Comment (nwoltman) [reply .eslintrc]: Whoops, must have hit backspace without realizing. Fixed now.

**Comment by amilajack** [comment]
Published this as `eslint-plugin-flowtype-errors@next` again. Would like some testing before bumping the minor semver

**Comment by amilajack** [comment]
Getting the following error:
```
messageMarkup.map is not a function
TypeError: messageMarkup.map is not a function
    at /Users/amila/Documents/Projects/falcon-core/node_modules/eslint-plugin-flowtype-errors/dist/collect.js:180:33
    at Array.map (<anonymous>)
    at collect (/Users/amila/Documents/Projects/falcon-core/node_modules/eslint-plugin-flowtype-errors/dist/collect.js:173:28)
    at Program (/Users/amila/Documents/Projects/falcon-core/node_modules/eslint-plugin-flowtype-errors/dist/index.js:83:46)
    at listeners.(anonymous function).forEach.listener (/Users/amila/Documents/Projects/falcon-core/node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/amila/Documents/Projects/falcon-core/node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (/Users/amila/Documents/Projects/falcon-core/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Users/amila/Documents/Projects/falcon-core/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (/Users/amila/Documents/Projects/falcon-core/node_modules/eslint/lib/util/node-event-generator.js:294:14)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by amilajack** [comment]
There are some cases where the `messageMarkup` is actually an object. It fails when it is equal to:
```
{ kind: 'UnorderedList',
  message: 
   [ { kind: 'Text', text: 'Cannot get ' },
     { kind: 'Code', text: 'exportOptions.tables' },
     { kind: 'Text', text: ' because:' } ],
  items: 
   [ [ [Object], [Object], [Object], [Object], [Object] ],
     [ [Object], [Object], [Object], [Object], [Object] ] ] }
```

**Comment by nwoltman** [comment]
@amilajack Could you post some code that will produce this error?

**Comment by jdmota** [comment]
https://github.com/facebook/flow/commit/28c8a3e5acd7ca9f17547ca38e54efd8475f9eb3

@nwoltman I think you just missed this

```js
type Message = MessageInline | MessageBlock;

type MessageBlock = MessageUnorderedList;

type MessageUnorderedList = {
  kind: 'UnorderedList',
  message: MessageInline,
  items: Array<Message>,
};
```

That explains why the message might be an object.

**Comment by jdmota** [comment]
I think it's easier if you just copy the types from facebook's commit message, to our code. That way, we won't forget anything.

**Comment by nwoltman** [comment]
@jdmota Thanks for pointing out that commit! I'll get to it tomorrow.

**Pull request merged**